client/src/c_console.cpp

```diff
@@ -852,10 +868,11 @@ static int C_PrintString(int printlevel, const char* color_code, const char* out
 
 	// [SL] the user's message color preference overrides the given color_code
 	// ...unless it's supposed to be formatted bold.
-	if (color_code && color_code[2] != '+' && printlevel >= 0 && printlevel < PRINTLEVELS)
+	static char printlevel_color_code[2];
+
+	if (color_code && color_code[1] != '+' && printlevel >= 0 && printlevel < PRINTLEVELS)
 	{
-		static char printlevel_color_code[4];
-		sprintf(printlevel_color_code, "\\c%c", 'a' + PrintColors[printlevel]);
+		sprintf(printlevel_color_code, "\034%c", 'a' + PrintColors[printlevel]);
```

Comment:
Here you are asking `sprintf` to write 3 characters (the two characters above plus the null terminator) into a 2 byte buffer.  In general, you should avoid the use of `sprintf` and prefer the use of `snprintf`.